### PR TITLE
Feature 2346 main v5.1 update templates

### LINKED ISSUE #2346: Update GitHub issue and pull request templates to reflect the current development workflow details
## Describe the Task ##
Some of the details in the GitHub issue and pull request templates are out of sync with the current development workflow. Update the contents of them to make them accurate.
Use the [MET issue](https://github.com/dtcenter/MET/issues/2659) as a guide.
Add "Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table." as a question in the Pull Request Template.
Also, update the applicable Issue Templates, adding to the checklist "For any new datasets, an entry to the [METplus Verification Datasets Guide](https://metplus.readthedocs.io/en/latest/Verification_Datasets/index.html)".  To the Pull Request templates, add to the checklist "Add any new datasets to the METplus Verification Datasets Guide](https://metplus.readthedocs.io/en/latest/Verification_Datasets/index.html)".

### Time Estimate ###
1 hour

### Sub-Issues ###
Consider breaking the task down into sub-issues.
None needed

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2793521

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by jprestop**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Reviewed the changes.  These changes are based off of the work from this [MET issue](https://github.com/dtcenter/MET/issues/2659) to improve the issue and pull request templates.  

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Review the changes.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[None needed]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Do these changes introduce new SonarQube findings? **[No]**</br>
If **yes**, please describe:

- [x] Please complete this pull request review by **[20240520]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/enhancement_request.md [hunk 1]
```diff
@@ -33,13 +33,14 @@ Consider breaking the enhancement down into sub-issues.
...
+- [ ] Review default **alert** labels
 - [ ] Select **component(s)**
 - [ ] Select **priority**
 - [ ] Select **requestor(s)**
 
-### Projects and Milestone ###
-- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
-- [ ] Select **Milestone** as the next official version or **Future Versions**
+### Milestone and Projects ###
+- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/enhancement_request.md]:
```suggestion
- [ ] Select **Milestone** as the next official version, **Consider for Next Release**, or **Backlog of Development Ideas**
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -65,20 +67,22 @@ Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
 - [ ] Add/update log messages for easier debugging.
 - [ ] Add/update unit tests.
 - [ ] Add/update documentation.
-- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
+- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
+- [ ] For any new datasets, an entry to the [METplus Verification Datasets Guide](https://metplus.readthedocs.io/en/latest/Verification_Datasets/index.html).
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/bug_report.md]: This may not be necessary for a bugfix issue since a fix to existing functionality is unlikely to add new datasets.
```suggestion
```

File: .github/ISSUE_TEMPLATE/new_feature_request.md [hunk 3]
```diff
@@ -37,13 +37,14 @@ Consider breaking the new feature down into sub-issues.
...
+- [ ] Review default **alert** labels
 - [ ] Select **component(s)**
 - [ ] Select **priority**
 - [ ] Select **requestor(s)**
 
-### Projects and Milestone ###
-- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
-- [ ] Select **Milestone** as the next official version or **Future Versions**
+### Milestone and Projects ###
+- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/new_feature_request.md]:
```suggestion
- [ ] Select **Milestone** as the next official version, **Consider for Next Release**, or **Backlog of Development Ideas**
```

File: .github/ISSUE_TEMPLATE/new_use_case.md [hunk 4]
```diff
@@ -45,14 +45,15 @@ Consider breaking the new feature down into sub-issues.
...
 - [ ] Select **component(s)**
 - [ ] Select **priority**
 - [ ] Select **requestor(s)**
 - [ ] Select **privacy**
 
-### Projects and Milestone ###
-- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
-- [ ] Select **Milestone** as the next official version or **Future Versions**
+### Milestone and Projects ###
+- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/new_use_case.md]:
```suggestion
- [ ] Select **Milestone** as the next official version, **Consider for Next Release**, or **Backlog of Development Ideas**
```

File: .github/ISSUE_TEMPLATE/sub-issue.md [hunk 5]
```diff
@@ -23,10 +23,11 @@ This is a sub-issue of #*List the parent issue number here*.
...
+- [ ] Review default **alert** labels
 - [ ] Select **component(s)**
 - [ ] Select **priority**
 - [ ] Select **requestor(s)**
 
-### Projects and Milestone ###
-- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
-- [ ] Select **Milestone** as the next official version or **Future Versions**
+### Milestone and Projects ###
+- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/sub-issue.md]:
```suggestion
- [ ] Select **Milestone** as the next official version, **Consider for Next Release**, or **Backlog of Development Ideas**
```

File: .github/ISSUE_TEMPLATE/task.md [hunk 6]
```diff
@@ -33,13 +33,14 @@ Consider breaking the task down into sub-issues.
...
+- [ ] Review default **alert** labels
 - [ ] Select **component(s)**
 - [ ] Select **priority**
 - [ ] Select **requestor(s)**
 
-### Projects and Milestone ###
-- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
-- [ ] Select **Milestone** as the next official version or **Future Versions**
+### Milestone and Projects ###
+- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/task.md]:
```suggestion
- [ ] Select **Milestone** as the next official version, **Consider for Next Release**, or **Backlog of Development Ideas**
```

File: .github/ISSUE_TEMPLATE/update_truth.md [hunk 7]
```diff
@@ -26,9 +26,9 @@ assignees: ''
 - [ ] Select **engineer(s)** or **no engineer** required
 - [ ] Select **scientist(s)** or **no scientist** required
 
-### Projects and Milestone ###
-- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
-- [ ] Select **Milestone** as the next official version or **Future Versions**
+### Milestone and Projects ###
+- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/update_truth.md]: Updating the truth data should never be added to the backlog since it should be completed ASAP. If updating truth for an existing release, the bugfix milestone should be used instead.
```suggestion
- [ ] Select **Milestone** as the next official version if updating truth data for the `develop` branch OR select next X.Y.Z bugfix version if updating truth data for a `main_vX.Y` branch
```

File: .github/ISSUE_TEMPLATE/update_truth.md [hunk 8]
```diff
@@ -26,9 +26,9 @@ assignees: ''
...
+- [ ] For the next official version, select the **METplus-Wrappers-X.Y.Z Development** project
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/update_truth.md]: If the truth data is being updated for an existing release, then the coordinated release support project should be selected instead. 
```suggestion
- [ ] If updating truth data for the `develop` branch, select the **METplus-Wrappers-X.Y.Z Development** project OR if updating truth for a `main_vX.Y` branch, select the **Coordinated METplus-X.Y Support** project
```

**Review by georgemccabe — CHANGES_REQUESTED** [review]
I made a few in-line suggestions. I recommend cherry-picking the commits of the changes you decide to make in this PR to apply them to the corresponding develop branch PR #2589

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (jprestop) [reply .github/ISSUE_TEMPLATE/bug_report.md]: @georgemccabe I thought the same thing at first, but then I thought there could be a case where the bug would have been discovered due to a new dataset.  Please confirm if you'd prefer I take this out or leave it in.  Thanks!

File: .github/ISSUE_TEMPLATE/update_truth.md (hunk 7 above)
Comment (jprestop) [reply .github/ISSUE_TEMPLATE/update_truth.md]: @georgemccabe Great point!  Clearly I wasn't using my brain for this one!🤪

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (georgemccabe) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I think the new dataset would be added with a different issue. If a bug was discovered, a new issue would be created to address the bug that is separate.

Comment (jprestop) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Given your comment, it sounds like you would prefer that I remove it, so I will do that.  Please let me know if I misunderstood.

**Review by georgemccabe — APPROVED** [review]
